HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc

```diff
@@ -284,7 +288,13 @@ void HLTScoutingEgammaProducer::produce(edm::StreamID sid, edm::Event& iEvent, e
       if (scRefFromTrk == scRef) {
         d0 = track.d0();
         dz = track.dz();
+        trackpt = track.pt();
+        tracketa = track.eta();
+        trackphi = track.phi();
         charge = track.charge();
+        auto const trackndof = track.ndof();
+        if (trackndof != 0)
+          trackchi2overndf = track.chi2() / trackndof;
```

Comment:
I don't think that tracks with zero degrees of freedom can arrive at this point. If so, then the `if` is useless here; if not, then probably a different  value of `trackchi2overndf` rather than the default 0 could be allowed (e.g. "-1."?)

Reply:
Okay, we could then do
```diff
-       if (trackndof != 0)
-          trackchi2overndf = track.chi2() / trackndof;
+      trackchi2overndf = (track.ndof() != 0) ? (track.chi2() / trackndof) : -1.;
+      break;
```
 - I understand the point about ndof, but since I don't know for sure, I would leave the check on zero.
 - The `break` potentially changes the results, if more than one match is possible (is it possible, @arsahasransu ?).